File: algorithms/0922.sort-array-by-parity-ii/solution.rs
impl Solution {
    pub fn sort_array_by_parity_ii(nums: Vec<i32>) -> Vec<i32> {
        let mut odds: Vec<i32> = vec![];
        let mut evens: Vec<i32> = vec![];
        for n in nums {
            if n % 2 == 0 {
                evens.push(n);
            } else {
                odds.push(n);
            }
        }

        let mut res: Vec<i32> = vec![];
        while !odds.is_empty() || !evens.is_empty() {
            res.push(evens.pop().unwrap());
            res.push(odds.pop().unwrap());
        }
        res
    }
}

File: algorithms/0922.sort-array-by-parity-ii/solution.go
func sortArrayByParityII(nums []int) []int {
	odd := []int{}
	even := []int{}
	for _, n := range nums {
		if n%2 == 0 {
			even = append(even, n)
		} else {
			odd = append(odd, n)
		}
	}

	res := []int{}
	for len(odd) > 0 || len(even) > 0 {
		if len(even) > 0 {
			res = append(res, even[len(even)-1])
			even = even[:len(even)-1]
		}
		if len(odd) > 0 {
			res = append(res, odd[len(odd)-1])
			odd = odd[:len(odd)-1]
		}
	}

	return res
}
